spec/requests/api/provision_requests_spec.rb

```diff
@@ -105,6 +105,140 @@ def app
     end
   end
 
+  context "AWS advanced provision requests" do
+    let(:aws_dialog)     { FactoryGirl.create(:miq_dialog_aws_provision) }
+    let(:ems)            { FactoryGirl.create(:ems_amazon_with_authentication) }
+    let(:template) do
+      FactoryGirl.create(:template_amazon, :name => "template1", :ext_management_system => ems)
+    end
+    let(:flavor) do
+      FactoryGirl.create(:flavor_amazon, :ems_id => ems.id, :name => 't2.small', :cloud_subnet_required => true)
+    end
+    let(:az)             { FactoryGirl.create(:availability_zone_amazon) }
+    let(:cloud_network1) { FactoryGirl.create(:cloud_network, :ems_id => ems.id, :enabled => true) }
+    let(:cloud_subnet1)  { FactoryGirl.create(:cloud_subnet, :cloud_network_id => cloud_network1.id) }
+
+    let(:provreq_body) do
+      {
+        "template_fields" => {"guid" => template.guid},
+        "requester"       => {"user_name" => api_config(:user)}
+      }
+    end
+
+    let(:provreq1_body) do
+      provreq_body.merge(
+        "vm_fields" => {
+          "vm_name"                     => "api_test_aws",
+          "instance_type"               => flavor.id,
+          "placement_auto"              => false,
+          "placement_availability_zone" => az.id,
+          "cloud_network"               => cloud_network1.id,
+          "cloud_subnet"                => cloud_subnet1.id,
+          "security_groups"             => 1,
+          "floating_ip_address"         => 1
```

Comment:
@romaintb You are not defining any objects for `security_groups ` or `floating_ip_address ` so I do not think you need to pass these values.